pkg/config/hcl_loader.go

```diff
@@ -0,0 +1,230 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+
+	backendpkg "github.com/kinvolk/lokomotive/pkg/backend"
+	"github.com/kinvolk/lokomotive/pkg/cluster/config"
+	"github.com/kinvolk/lokomotive/pkg/components"
+)
+
+// HCLLoader represents loading the HCL configuration provided by the user
+type HCLLoader struct {
+	ConfigPath    string
+	VariablesPath string
+}
+
+// Load loads the HCL files provided by the user and parses them into an
+// instance of LokomoticeConfig
+func (c *HCLLoader) Load() (*config.LokomotiveConfig, hcl.Diagnostics) {
+	configFiles, diags := loadHCLFiles(c, "lokocfg")
+	if diags.HasErrors() {
+		return nil, diags
+	}
+	variablesFile, diags := loadHCLFiles(c, "vars")
+	if diags.HasErrors() {
+		return nil, diags
+	}
+	hclConfig, diags := ParseHCLFiles(configFiles, variablesFile)
+	if diags.HasErrors() {
+		return nil, diags
+	}
+
+	return parseToLokomotiveConfig(hclConfig)
+}
+
+// parseToLokomotiveConfig parses the Config instance to LokomotiveConfig
```

Comment:
All comments should end with `.`.
```suggestion
// parseToLokomotiveConfig parses the Config instance to LokomotiveConfig.
```